src/main/java/net/snowflake/client/core/client_config/SFClientConfigParser.java

```diff
@@ -0,0 +1,92 @@
+package net.snowflake.client.core.client_config;
+
+import static net.snowflake.client.jdbc.SnowflakeUtil.systemGetEnv;
+import static net.snowflake.client.jdbc.SnowflakeUtil.systemGetProperty;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import net.snowflake.client.jdbc.SnowflakeDriver;
+
+public class SFClientConfigParser {
+  public static final String SF_CLIENT_CONFIG_FILE_NAME = "sf_client_config.json";
+  public static final String SF_CLIENT_CONFIG_ENV_NAME = "SF_CLIENT_CONFIG_FILE";
+
+  /**
+   * Construct SFClientConfig from client config file passed by user. This method searches the
+   * config file in following order: 1. configFilePath param which is read from connection URL or
+   * connection property. 2. Environment variable: SF_CLIENT_CONFIG_FILE containing full path to
+   * sf_client_config file. 3. Searches for default config file name(sf_client_config.json under the
+   * driver directory from where the driver gets loaded. 4. Searches for default config file
+   * name(sf_client_config.json) under user home directory 5. Searches for default config file
+   * name(sf_client_config.json) under tmp directory.
+   *
+   * @param configFilePath SF_CLIENT_CONFIG_FILE parameter read from connection URL or connection
+   *     properties
+   * @return SFClientConfig
+   */
+  public static SFClientConfig loadSFClientConfig(String configFilePath) {
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    File configFile = null;
+    try {
+      if (configFilePath != null && !configFilePath.isEmpty()) {
+        // 1. Try to read the file at  configFilePath.
+        configFile = new File(configFilePath);
```

Comment:
Can we take path in a variable and do configFile and next line at one place in this method? Also we need to log which path we take as there are many options, so it will help support to locate the file.

Reply:
The main reason why I didn't write to log statements is that the logger may not be initialized with client config values. If we keep log statements here, they might end up showing in the default location or they won't be displayed at all if the `logging.properties` file isn't defined.  Not sure if adding logging statement here would help.